builder/simplify_test.cc

```diff
@@ -64,10 +64,13 @@ void test_simplify(const stmt& test, const stmt& expected) {
 }
 
 TEST(simplify, basic) {
+  test_simplify(expr() == 1, expr() == 1);
```

Comment:
This is a surprising result; I would have expected `expr() == 1` to always simplify to `false`

Reply:
I clearly don't understand, then, `expr()` is an undefined expression and it seems like comparing it to 1 should always simplify into a constant false

Reply:
That implies that the result of that expression is *always* false, but it isn't, it's undefined, which includes possibly 1.

I have not yet made any explicit choice, here, this test is just verifying the same behavior we've always had by accident. If we want to actually define this behavior to be something different, we can do that (but I think your proposed behavior would require some debugging). IMO `prove_true(expr() == 1)` and `prove_false(expr() == 1` should both return false, but your expected behavior wouldn't do that.